kitsune/sumo/templatetags/jinja_helpers.py

```diff
@@ -226,16 +234,18 @@ def datetimeformat(context, value, format='shortdatetime'):
     else:
         new_value = value
 
-    if 'timezone' not in request.session:
-        if request.user.is_authenticated():
-            try:
-                convert_tzinfo = (Profile.objects.get(user=request.user).timezone or
-                                  default_tzinfo)
-            except (Profile.DoesNotExist, AttributeError):
-                pass
-        request.session['timezone'] = convert_tzinfo
-    else:
-        convert_tzinfo = request.session['timezone']
+    try:
+        if 'timezone' not in request.session:
+            if request.user.is_authenticated():
+                user_timezone = get_user_timezone(request.user)
+                if user_timezone is not None:
+                    convert_tzinfo = user_timezone
+            request.session['timezone'] = convert_tzinfo
+        else:
+            convert_tzinfo = request.session['timezone']
+
+    except AttributeError:
```

Comment:
I worry that this try/except is too broad. I'd rather see something like `if hasattr(request, 'session')`, which would be very specific. Catching generic `AttributeError`s, in particular, can have unwanted side effects as it can hide programmer errors, like misspelling a variable name.